src/main/scala/beam/utils/StuckFinderHelper.scala

```diff
@@ -0,0 +1,54 @@
+package beam.utils
+
+import java.util.{Comparator, PriorityQueue}
+
+import scala.collection.mutable
+
+case class ValueWithTime[T](value: T, time: Long)
+
+class StuckFinderHelper[K] {
+
+  object Comparator extends Comparator[ValueWithTime[K]] {
+    def compare(o1: ValueWithTime[K], o2: ValueWithTime[K]): Int = {
+      o1.time.compare(o2.time)
+    }
+  }
+
+  private[this] val pq = new PriorityQueue[ValueWithTime[K]](Comparator)
```

Comment:
Any particular reason you chose Java's PQ over Scala's? Just curious.

Reply:
[Scala Priority Queue](https://www.scala-lang.org/api/2.12.0/scala/collection/mutable/PriorityQueue.html) does not have an API to remove particular element from the queue

Reply:
I guess that begs the question of why that is needed? It defeats the purpose of the PQ otherwise? Additionally, the only place I see you using it is with `removeByKey`, which isn't needed after the poll and the other usage I guess will depend on the open question in BeamAgentScheduler that you pinged Colin on. So it can be re-addressed once that is cleared up.